server/integration/testsuite/src/test/java/org/infinispan/server/test/cache/container/OffHeapCompatibilityIT.java

```diff
@@ -0,0 +1,105 @@
+package org.infinispan.server.test.cache.container;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.entity.ContentType;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.infinispan.arquillian.core.InfinispanResource;
+import org.infinispan.arquillian.core.RemoteInfinispanServer;
+import org.infinispan.arquillian.core.RemoteInfinispanServers;
+import org.infinispan.arquillian.core.RunningServer;
+import org.infinispan.arquillian.core.WithRunningServer;
+import org.infinispan.client.hotrod.Flag;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.server.test.util.ITestUtils;
+import org.infinispan.server.test.util.RemoteInfinispanMBeans;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Compatibility mode tests with off-heap data container
+ *
+ * @author vjuranek
+ * @since 9.2
+ */
+@RunWith(Arquillian.class)
+@WithRunningServer({@RunningServer(name = OffHeapCompatibilityIT.CONFIG_NAME)})
+public class OffHeapCompatibilityIT {
```

Comment:
OFF_HEAP and compatibility mode is (no pun intended) incompatible, IMO it should throw an error if someone configured the cache with both.

Reply:
well, it doesn't, it throws exception later on, see [ISPN-8445](https://issues.jboss.org/browse/ISPN-8445). Also, AFAICT the incompatibility is not documented. 
Btw: why these two are not compatible, both store the data as raw bytes, so I'd expect that they can be used together.

Reply:
> Btw: why these two are not compatible

In order to interop between remote endpoints and embedded, compat mode internally relies on storing content in "unmarshalled form", so in theory even code running inside the server's JVM would be able to "read" objects without having to deal with marshalling. But OFF_HEAP has a hard requirement on storing byte[ ], so there's a conflict here.

>  both store the data as raw bytes, so I'd expect that they can be used together.

+100, an OFF_HEAP cache in the server should be able to work together with Rest. Have you tried to do so without using compat mode? It should work...

Reply:
> OFF_HEAP cache in the server should be able to work together with Rest. Have you tried to do so without using compat mode? It should work...

yes, it works, test for it is actually [part](https://github.com/infinispan/infinispan/pull/5544/files#diff-60226fbe2e1fa9ef1e73352fa706a3e0R140) of this PR

Reply:
I have fixed the issue with OFF-HEAP and compatibility. The issue was that the memory configuration validate method was never invoked :(

https://github.com/infinispan/infinispan/pull/5590